Content.Server/_Scp/Scp106/Systems/Scp106System.cs

```diff
@@ -216,19 +345,85 @@ public override void Update(float frameTime)
     {
         base.Update(frameTime);
 
-        var query = AllEntityQuery<Scp106Component>();
+        var queryScp106 = AllEntityQuery<Scp106Component>();
 
-        while(query.MoveNext(out var uid, out var comp))
+        while(queryScp106.MoveNext(out var scp106Uid, out var scp106Component))
         {
-            comp.Accumulator += frameTime;
+            scp106Component.Accumulator += frameTime;
+
+            if (scp106Component.Accumulator < 1)
+                continue;
 
-            if (comp.Accumulator < 180)
+            scp106Component.BackroomsAccumulator += 1;
+            scp106Component.Essence += 1;
+
+            scp106Component.Accumulator -= 1;
+            var HumansInBackrooms = 0;
+
+            if (scp106Component.BackroomsAccumulator < 60)
+            {
+                Dirty(scp106Uid, scp106Component);
                 continue;
+            }
+
+            scp106Component.BackroomsAccumulator = 0;
+
+            var queryHumans = AllEntityQuery<HumanoidAppearanceComponent, MobStateComponent>();
+
+            while (queryHumans.MoveNext(out var humanUid, out var _, out var mobStateComponent))
+            {
+                if (HasComp<Scp106BackRoomMapComponent>(Transform(humanUid).MapUid)
+                    && mobStateComponent.CurrentState == MobState.Alive)
+                {
+                    HumansInBackrooms += 1;
+
+                    if (HumansInBackrooms >= 10)
+                    {
+                        // Тут должна быть логика события, например добавление особой абилки
+                        Dirty(scp106Uid, scp106Component);
+                    }
+                }
+            }
 
-            comp.Accumulator -= 180;
-            comp.AmoutOfPhantoms += 1;
-            Dirty(uid, comp);
+            if (scp106Component.AnnouncementAccumulator < 600)
+            {
+                scp106Component.AnnouncementAccumulator += 60;
+                Dirty(scp106Uid, scp106Component);
+                continue;
+            }
+
+            if (HumansInBackrooms >= 10)
+            {
+                scp106Component.AnnouncementAccumulator = 0;
+
+                var boo = new GhostBooEvent();
+                RaiseLocalEvent(scp106Uid, boo, true);
+
+                var statusEffectQuery = EntityQueryEnumerator<StatusEffectsComponent>();
```

Comment:
```suggestion
                var statusEffectQuery = EntityQueryEnumerator<HumanoidAppearanceComponent, StatusEffectsComponent>();
```

Я уверен, что тебе не нужно, чтобы все сцп и даже корова в блоке Д тряслись от накопления 10 человек в измерении 106